src/matter/cluster/server/OperationalCredentialsServer.ts

```diff
@@ -109,16 +110,18 @@ export const OperationalCredentialsClusterHandler: (conf: OperationalCredentials
     removeFabric: async ({ request: {fabricIndex}, session }) => {
         const device = session.getContext();
 
-        try {
+        const status = tryCatch(() => {
             device.removeFabric(fabricIndex);
-        } catch {
-            return { status: OperationalCertStatus.InvalidFabricIndex };
-        }
+            return OperationalCertStatus.Success;
+        },
+        FabricNotFoundError, OperationalCertStatus.InvalidFabricIndex);
 
-        // TODO persist fabrics
-        // TODO: depending on cases destroy the secure session and delete all data!
+        if (status === OperationalCertStatus.Success) {
+            // TODO persist fabrics
```

Comment:
You can move this after device.removeFabric(fabricIndex);
No need to had a separate if block